zgw/catalogi-api/src/main/kotlin/com/ritense/catalogiapi/CatalogiApiPlugin.kt

```diff
@@ -121,195 +139,246 @@ class CatalogiApiPlugin(
+                    authenticationPluginConfiguration,
+                    url,
+                    RoltypeRequest(
+                        zaaktype = zaakTypeUrl,
+                        page = currentPage++
+                    )
                 )
-            )
-            results.addAll(currentResults.results)
-        } while (currentResults?.next != null)
+                results.addAll(currentResults.results)
+            } while (currentResults?.next != null)
 
-        return results
+            return results
+        }
     }
 
     fun getStatustypen(zaakTypeUrl: URI): List<Statustype> {
-        var currentPage = 1
-        var currentResults: Page<Statustype>?
-        val results = mutableListOf<Statustype>()
-
-        do {
-            logger.debug { "Getting page of statustypen, page $currentPage for zaaktype $zaakTypeUrl" }
-            currentResults = client.getStatustypen(
-                authenticationPluginConfiguration,
-                url,
-                StatustypeRequest(
-                    zaaktype = zaakTypeUrl,
-                    page = currentPage++
+        withLoggingContext(CATALOGI_API.STATUSTYPE to zaakTypeUrl.toString()) {
+            var currentPage = 1
+            var currentResults: Page<Statustype>?
+            val results = mutableListOf<Statustype>()
+
+            do {
+                logger.debug { "Getting page $currentPage of statustypen for zaaktype $zaakTypeUrl" }
+                currentResults = client.getStatustypen(
+                    authenticationPluginConfiguration,
+                    url,
+                    StatustypeRequest(
+                        zaaktype = zaakTypeUrl,
+                        page = currentPage++
+                    )
                 )
-            )
-            results.addAll(currentResults.results)
-        } while (currentResults?.next != null)
+                results.addAll(currentResults.results)
+            } while (currentResults?.next != null)
 
-        return results
+            return results
+        }
     }
 
     fun getStatustype(statusTypeUrl: URI): Statustype {
-        return client.getStatustype(authenticationPluginConfiguration, url, statusTypeUrl)
+        withLoggingContext(CATALOGI_API.STATUSTYPE to statusTypeUrl.toString()) {
+            logger.debug { "Getting Statustype for URL: $statusTypeUrl" }
+            return client.getStatustype(authenticationPluginConfiguration, url, statusTypeUrl)
+        }
     }
 
     fun getStatustypeByOmschrijving(zaakTypeUrl: URI, omschrijving: String): Statustype {
-        return getStatustypen(zaakTypeUrl)
-            .singleOrNull { it.omschrijving.equals(omschrijving, ignoreCase = true) }
-            ?: throw StatustypeNotFoundException("With 'omschrijving': '$omschrijving'")
+        withLoggingContext(
+            CATALOGI_API.STATUSTYPE to zaakTypeUrl.toString(),
+            CATALOGI_API.STATUSTYPE to omschrijving
```

Comment:
Both keys are the same, but i think we don't need the omschrijving as MDC variable.